### PR TITLE
Add kfdef CRD back

### PR DESCRIPTION
**Pull request by zdtsw**
## Description
add back kfdef CRD (cherry-pick from master branch of old operator)  this is still required by odh-dashboard

## How Has This Been Tested?
when it is not there, odh-dashboard pod error
```
{"level":30,"time":1688972181096,"pid":17,"hostname":"odh-dashboard-58fb784664-8sd5f","reqId":"req-1vc","res":{"statusCode":200},"responseTime":0.2644881010055542,"msg":"request completed"}
{"level":50,"time":1688972185022,"pid":17,"hostname":"odh-dashboard-58fb784664-8sd5f","err":{"type":"HttpError","message":"HTTP request failed","stack":"HttpError: HTTP request failed\n    at Request._callback (/usr/src/app/backend/node_modules/@kubernetes/client-node/dist/gen/api/customObjectsApi.js:1482:40)\n    at self.callback (/usr/src/app/backend/node_modules/request/request.js:185:22)\n    at Request.emit (node:events:513:28)\n    at Request.<anonymous> (/usr/src/app/backend/node_modules/request/request.js:1154:10)\n    at Request.emit (node:events:513:28)\n    at IncomingMessage.<anonymous> (/usr/src/app/backend/node_modules/request/request.js:1076:12)\n    at Object.onceWrapper (node:events:627:28)\n    at IncomingMessage.emit (node:events:525:35)\n    at endReadableNT (node:internal/streams/readable:1359:12)\n    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)","response":{"statusCode":404,"body":"404 page not found\n","headers":{"audit-id":"07900383-0b5e-4ef4-8c47-6114a5eeb7e4","cache-control":"no-cache, private","content-type":"text/plain; charset=utf-8","x-content-type-options":"nosniff","x-kubernetes-pf-flowschema-uid":"9546f919-28af-439f-9c13-33af617d0837","x-kubernetes-pf-prioritylevel-uid":"2e3787dd-b27b-40ae-8d1f-464021030194","date":"Mon, 10 Jul 2023 06:56:25 GMT","content-length":"19","connection":"close"},"request":{"uri":{"protocol":"https:","slashes":true,"auth":null,"host":"172.30.0.1:443","port":"443","hostname":"172.30.0.1","hash":null,"search":null,"query":null,"pathname":"/apis/kfdef.apps.kubeflow.org/v1/namespaces/opendatahub/kfdefs","path":"/apis/kfdef.apps.kubeflow.org/v1/namespaces/opendatahub/kfdefs","href":"https://172.30.0.1:443/apis/kfdef.apps.kubeflow.org/v1/namespaces/opendatahub/kfdefs"},"method":"GET","headers":{"Accept":"application/json","Authorization":"Bearer eyJhbGciOiJSUzI1NiIsImtpZCI6IlQ5RXNsTDhhN3UyLVlEMW1Rb21ESXB0UXdVRmpEOTRDVjZmR0VMZjZUdWsifQ.eyJhdWQiOlsiaHR0cHM6Ly9yaC1vaWRjLnMzLnVzLWVhc3QtMS5hbWF6b25hd3MuY29tLzIzYzczNHN0M3BuN2wxNjdtcTk3ZDBvdDg4NDhsZ3JsIl0sImV4cCI6MTcyMDUwNTU0MywiaWF0IjoxNjg4OTY5NTQzLCJpc3MiOiJodHRwczovL3JoLW9pZGMuczMudXMtZWFzdC0xLmFtYXpvbmF3cy5jb20vMjNjNzM0c3QzcG43bDE2N21xOTdkMG90ODg0OGxncmwiLCJrdWJlcm5ldGVzLmlvIjp7Im5hbWVzcGFjZSI6Im9wZW5kYXRhaHViIiwicG9kIjp7Im5hbWUiOiJvZGgtZGFzaGJvYXJkLTU4ZmI3ODQ2NjQtOHNkNWYiLCJ1aWQiOiIwZmU2MzhjZS01MzE3LTQ3ZWEtYjdlZS1iMmJmZDI1MTAwN2QifSwic2VydmljZWFjY291bnQiOnsibmFtZSI6Im9kaC1kYXNoYm9hcmQiLCJ1aWQiOiJlNTg4YjJlNy1lNjZhLTQ2M2ItYjMzYS0zMmVmMDQ3YjU1ZjkifSwid2FybmFmdGVyIjoxNjg4OTczMTUwfSwibmJmIjoxNjg4OTY5NTQzLCJzdWIiOiJzeXN0ZW06c2VydmljZWFjY291bnQ6b3BlbmRhdGFodWI6b2RoLWRhc2hib2FyZCJ9.Zup-mazVL_oBLxloey-mTWeMkW6u_TWph17veHo38QEgSjegPpClCpqD-64QxOdsTL6dcrO16U9AysXeJ9PcDUb_OpUwK-c0XadABHBxcjpmQKGu_HptozERuKhoVoNssRV7JlaOCu7F3XsFC3BQ9gr94_3CuExUXBPoDxcwhXFA-ofPcBAk7VJ62GiVN9maY7PCVhmloGVL9fa1GSRYLFEMU11RhLoANQggdoUoJOwSd0CMctzgZQuLGf5fR05ObvMZbqapDozg1FTGdIOnkzO9wIoJzPt2hXTJFIoidZKLq_1x8iBuiOnFSX70pVjGSMxLdgKwVJ897aJy-ojBQU5C_HSmvqatWF9mecCRyjLQjhBP00vBhRtVCdEElcTxNcrfWGJxGa93gVHty3AXAPINWlNvlsB7G7I3XuvdQbwuUxwVDh33szDKyWqLho5fLDXP6rVX0eAcC9KYZgVx3gH3KicrcGhHrkNRxl0C4HyKlwClAYu3AOcL1WWHCWl2B5BfGDWVQoSTx9PbFtt7UyDJSqyVqJ7sAUwxhy97CcwX6i3uhoxfTKdnXz0kOd8I30vurVWSkG7lpW_JSaeL1BaYuO1bn2FQ3wJ_ntomu7T2UIWTJz7GBo9X3l6RVwwRYRUXdKcDH6SpoeOOBfecNNF9_B_-soL7j-qO0C84798"}}},"body":"404 page not found\n","statusCode":404,"name":"HttpError"},"msg":"failed to get kfdefs"}
{"level":30,"time":1688972185024,"pid":17,"hostname":"odh-dashboard-58fb784664-8sd5f","msg":"Pipelines feature enabled, computing operator state..."}
{"level":30,"time":1688972185041,"pid":17,"hostname":"odh-dashboard-58fb784664-8sd5f","msg":"Pipeline Operator status: false"}
{"level":30,"time":1688972211096,"pid":17,"hostname":"odh-dashboard-58fb784664-8sd5f","reqId":"req-1vd","req":{"method":"GET","url":"/api/health","hostname":"10.128.1.114:8080","remoteAddress":"10.128.0.2","remotePort":50262},"msg":"incoming request"}
```
after add it
```
{"level":30,"time":1688975245496,"pid":17,"hostname":"odh-dashboard-58fb784664-kwxw9","reqId":"req-h","res":{"statusCode":200},"responseTime":0.565019965171814,"msg":"request completed"}
{"level":30,"time":1688975275495,"pid":17,"hostname":"odh-dashboard-58fb784664-kwxw9","reqId":"req-i","req":{"method":"GET","url":"/api/health","hostname":"10.128.1.247:8080","remoteAddress":"10.128.0.2","remotePort":39376},"msg":"incoming request"}
{"level":30,"time":1688975275496,"pid":17,"hostname":"odh-dashboard-58fb784664-kwxw9","reqId":"req-i","res":{"statusCode":200},"responseTime":0.49156200885772705,"msg":"request completed"}
{"level":30,"time":1688975305495,"pid":17,"hostname":"odh-dashboard-58fb784664-kwxw9","reqId":"req-j","req":{"method":"GET","url":"/api/health","hostname":"10.128.1.247:8080","remoteAddress":"10.128.0.2","remotePort":45898},"msg":"incoming request"}
{"level":30,"time":1688975305496,"pid":17,"hostname":"odh-dashboard-58fb784664-kwxw9","reqId":"req-j","res":{"statusCode":200},"responseTime":0.4821920394897461,"msg":"request completed"}
{"level":30,"time":1688975335495,"pid":17,"hostname":"odh-dashboard-58fb784664-kwxw9","reqId":"req-k","req":{"method":"GET","url":"/api/health","hostname":"10.128.1.247:8080","remoteAddress":"10.128.0.2","remotePort":40636},"msg":"incoming request"}
{"level":30,"time":1688975335496,"pid":17,"hostname":"odh-dashboard-58fb784664-kwxw9","reqId":"req-k","res":{"statusCode":200},"responseTime":0.3167020082473755,"msg":"request completed"}
{"level":30,"time":1688975339394,"pid":17,"hostname":"odh-dashboard-58fb784664-kwxw9","msg":"Pipelines feature enabled, computing operator state..."}
{"level":30,"time":1688975339432,"pid":17,"hostname":"odh-dashboard-58fb784664-kwxw9","msg":"Pipeline Operator status: false"}
{"level":30,"time":1688975365495,"pid":17,"hostname":"odh-dashboard-58fb784664-kwxw9","reqId":"req-l","req":{"method":"GET","url":"/api/health","hostname":"10.128.1.247:8080","remoteAddress":"10.128.0.2","remotePort":42096},"msg":"incoming request"}
{"level":30,"time":1688975365496,"pid":17,"hostname":"odh-dashboard-58fb784664-kwxw9","reqId":"req-l","res":{"statusCode":200},"responseTime":0.43930697441101074,"msg":"request completed"}
```
**UPDATE**
test with new catsrc image from last commit:
do not see "kfdef" which can be used to create instance by user
Screenshot from 2023-07-17 15-13-44

Screenshot from 2023-07-17 15-14-20
kfdef CRD created 
Screenshot from 2023-07-17 15-14-00

Screenshot from 2023-07-17 15-14-31

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by VaishnaviHire** [comment]
Created https://github.com/opendatahub-io/odh-dashboard/issues/1491 

@zdtsw Can you create a corresponding issue to this PR and set it as blocker for the release ?

**Comment by zdtsw** [comment]
> Created [opendatahub-io/odh-dashboard#1491](https://github.com/opendatahub-io/odh-dashboard/issues/1491)
> 
> @zdtsw Can you create a corresponding issue to this PR and set it as blocker for the release ?

on my way

**Comment by zdtsw** [comment]
> @zdtsw Is the Dashboard route working after this change?

i was testing this change (after 2nd commit) still i cannot get Route work

File: config/crd/bases/kfdef.apps.kubeflow.org_kfdefs.yaml [hunk 1]
```diff
@@ -0,0 +1,187 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
```
Comment (VaishnaviHire) [inline config/crd/bases/kfdef.apps.kubeflow.org_kfdefs.yaml:2]: This crd should not be added default `crd` folder.

You can create custom folder. Eg. `additional/kfdef_crd.yaml`

Comment (zdtsw) [reply config/crd/bases/kfdef.apps.kubeflow.org_kfdefs.yaml]: done, as `config/crd/additionals/kfdef.apps.kubeflow.org_kfdefs.yaml`

**Review by VaishnaviHire — APPROVED** [review]
/lgtm

**Pull request merged**